lib/src/data/group_class.dart

```diff
@@ -0,0 +1,47 @@
+import "utils.dart";
+
+enum GroupStatus {
+  openToAll,
+  openByRequest,
+  private,
+}
+
+class Group {
+  final String id;
+  final String name;
+  final List<String> members; 
+  int? memberLimit;
+  GroupStatus status;
+
+  Group({
+    required this.id,
+    required this.name,
+    required this.members,
+    required this.status,
+    this.memberLimit, 
```

Comment:
```suggestion
    required this.memberLimit, 
```

Required in the _constructor_ doesn't necessarily mean required in the UI. The UI can have a default value like 
- [ ] Limit members to ___

And we can leave that checkbox unticked by default. But in the code itself, the value would still explicitly be `null`, so we won't need a default. There's little difference between removing `required` here and expecting null to be explicitly passed in, but by using `required` you're forcing the frontend guy who will work on that page later to remember about the member limit option. If they forget it, they will get a compile-time error, so we prefer using `required` unless there's a really obvious default value.